taichi/transforms/ir_printer.cpp

```diff
@@ -34,6 +34,18 @@ std::string block_dim_info(int block_dim) {
          (block_dim == 0 ? "adaptive" : std::to_string(block_dim)) + " ";
 }
 
+std::string escape_new_line(const std::string &s) {
+  std::stringstream ss;
+  for (char c : s) {
+    if (c == '\n') {
+      ss << '\\' << 'n';
+    } else {
+      ss << c;
+    }
+  }
+  return ss.str();
+}
+
```

Comment:
Then what about '\\', '\r' and '\t'? Then won't display correctly too.

Just FYI:
https://github.com/taichi-dev/taichi/blob/ee0281f61278ebd429593391231e432b3c3fd1fc/taichi/backends/cc/cc_utils.h#L13-L39
Maybe we could make this a common utility in some common header?

Reply:
yeah, that SGTM

Reply:
Done